### ISSUE TITLE
NullPointerException in onRendererFilterChanged 

### ISSUE THREAD

**Issue by usmanrana07** [natario1/CameraView#762]
### Describe the bug
When i want to capture an image for first time and as i tap Cam icon after allow the permissions app is being crashed.

- CameraView version: *2.6.1*
- Device / Android version: *Samsung A30, API 28*

### Logs

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.otaliastudios.cameraview.internal.egl.EglViewport.setFilter(com.otaliastudios.cameraview.filter.Filter)' on a null object reference
       at com.otaliastudios.cameraview.picture.SnapshotGlPictureRecorder.onRendererFilterChanged + 134(SnapshotGlPictureRecorder.java:134)
       at com.otaliastudios.cameraview.picture.SnapshotGlPictureRecorder$1.onRendererFilterChanged + 98(SnapshotGlPictureRecorder.java:98)
       at com.otaliastudios.cameraview.preview.GlCameraPreview$2.run + 305(GlCameraPreview.java:305)
       at android.opengl.GLSurfaceView$GLThread.guardedRun + 1490(GLSurfaceView.java:1490)
       at android.opengl.GLSurfaceView$GLThread.run + 1262(GLSurfaceView.java:1262)
```

Following is the xml:

 

```
<com.otaliastudios.cameraview.CameraView
                android:id="@+id/camera"
                android:layout_width="match_parent"
                android:layout_height="match_parent"
                flashMode="@{viewModel.flashType}"
                setFacing="@{viewModel.facingCamera}"
                app:cameraExperimental="false"
                app:cameraEngine="camera2"
                app:cameraPreview="glSurface"
                app:cameraPlaySounds="false"
                app:cameraGrid="off"
                app:cameraAudio="off"
                app:cameraGestureTap="autoFocus"
                app:cameraGestureLongTap="none"
                app:cameraGesturePinch="zoom"
                app:cameraGestureScrollHorizontal="none"
                app:cameraGestureScrollVertical="none"
                app:cameraMode="picture"
                app:cameraUseDeviceOrientation="false"
                app:cameraRequestPermissions="false"
                app:cameraAutoFocusMarker="@string/cameraview_default_autofocus_marker" />

```

**Comment by brnogz** [comment]
@natario1 I have the same issue, do I need to create new issue ?  There is nothing I can add but it is trending crash on crashlytics.

**Comment by natario1** [comment]
I think this might be fixed already @brnogz . If the issue persists with next version feel free to open a new issue!

**Comment by VolodymyrMachekhin** [comment]
Got this issue in 2.6.2. Also what I see in logs is `app passed NULL surface`. 
Issue appears only after granting permissions.
